examples/nemo_run/prune_distill/nemo_prune_kd_flow.py

```diff
@@ -14,255 +14,315 @@
             seq_length=SEQUENCE_LENGTH,
         )
     else:
         tokenizer = run.Config(
             get_nmt_tokenizer,
             library="huggingface",
-            model_name=args.tokenizer_hf,
-            chat_template=_read_chat_template(args.chat_template) if args.chat_template else None,
+            model_name=args.hf_tokenizer,
         )
         data = run.Config(
-            ChatDataModule,
-            dataset_root=args.data_path,
+            PreTrainingDataModule,
+            paths=[f"{args.data_dir}/part_{i}_text_document" for i in SUBSET_IDX],
             seq_length=SEQUENCE_LENGTH,
             tokenizer=tokenizer,
-            global_batch_size=GBS,
-            micro_batch_size=MBS,
-            use_hf_tokenizer_chat_template=True,
+            global_batch_size=DISTILL_GBS,
+            micro_batch_size=DISTILL_MBS,
+            split="99,1,0",
         )
 
-    # 1. Process data and Import and save initial NeMo checkpoint
-    lima_data = run.Script("process_lima.py", entrypoint="python")
-
-    initial_model_out = f"{args.experiment_name}/{model_name}_initial"
+    # 1. Import and save initial NeMo checkpoint
+    initial_model_out = f"{exp_dir}/{model_name}_initial"
+    model_module = getattr(llm, model_name)
     import_model = run.Partial(
         llm.import_ckpt,
         model=model_module.model(),
-        source=f"hf://{args.model_hf}",
+        source=f"hf://{args.model_name}",
         output_path=initial_model_out,
+        overwrite=True,
     )
 
-    # 2. Finetune
-    finetune = get_finetune_recipe(args.recipe)
-    finetune.tokenizer = "data"
-    finetune.data = data
-    finetune.resume.restore_config.path = initial_model_out
-    finetune.log.log_dir = f"{args.experiment_name}/finetune_log_dir"
-    finetune.trainer.strategy.tensor_model_parallel_size = GPUS
-    finetune.trainer.max_steps = FINETUNE_STEPS
-    finetune.trainer.val_check_interval = VAL_INTERVAL
-    finetune.trainer.limit_val_batches = VAL_BATCHES
-
-    # 3. Prune to obtain student
+    # 2. Prune to obtain student
     prune_data = data.clone()
+    prune_data.micro_batch_size = PRUNE_MBS
     prune_data.global_batch_size = prune_data.micro_batch_size
-    finetuned_teacher_path = run.Config(get_most_recent_ckpt, finetune.log.log_dir)
-    pruned_model_out = f"{args.experiment_name}/{model_name}_pruned"
+    pruned_model_out = f"{exp_dir}/{model_name}_pruned"
     prune = prune_recipe(
-        nemo_checkpoint=finetuned_teacher_path,
+        nemo_checkpoint=initial_model_out,
         save_path=pruned_model_out,
     )
-    prune.pruning_config.target_hidden_size = args.prune_target_hidden_size
-    prune.pruning_config.target_ffn_hidden_size = args.prune_target_hidden_size * 3
-    prune.devices = GPUS
-    prune.pp_size = GPUS
+    prune.tokenizer_path = args.hf_tokenizer
+    prune.pruning_config.target_num_layers = args.prune_target_num_layers
+    prune.devices = 1
+    prune.pp_size = 1
```

Comment:
What are your thoughts on having a yaml config file with all these parameters? We can provide a config for Qwen3-8B depth prune but users can change it to what they like and they will have one single place to see all configurable parameters? We can also allow for different values for Prune PP and Distill TP (default to 1, 8 as you have here)